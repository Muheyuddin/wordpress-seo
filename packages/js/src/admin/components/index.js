export { default as RouteErrorFallback } from "./route-error-fallback";
export { default as SidebarLayout } from "./sidebar-layout";
export { default as Topbar } from "./topbar";
