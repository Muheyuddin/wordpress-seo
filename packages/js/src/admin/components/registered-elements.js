import { Fragment } from "@wordpress/element";
import { useRootContext } from "@yoast/ui-library";

/**
 * @returns {JSX.Element} The element.
 */
const RegisteredElements = () => {
	const { elements } = useRootContext();

	return elements.map( ( { key, value } ) => <Fragment key={ key }>{ value }</Fragment> );
};

export default RegisteredElements;
