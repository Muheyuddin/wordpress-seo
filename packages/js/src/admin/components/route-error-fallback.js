import { useCallback } from "@wordpress/element";
import { __ } from "@wordpress/i18n";
import { Alert, Button, Title } from "@yoast/ui-library";
import PropTypes from "prop-types";

/**
 * @param {Object} error The error instance.
 * @returns {JSX.Element} The element.
 */
const RouteErrorFallback = ( { error } ) => {
	const handleRefreshClick = useCallback( () => window?.location?.reload(), [] );

	return (
		<div role="alert" className="yst-max-w-screen-sm yst-p-8 yst-space-y-4">
			<Title>{ __( "Something went wrong. An unexpected error occurred.", "wordpress-seo" ) }</Title>
			<p>{ __( "We're very sorry, but it seems like the following error has interrupted our application:", "wordpress-seo" ) }</p>
			<Alert variant="error">{ error?.message || "Undefined error message." }</Alert>
			<p>{ __( "Unfortunately, this means that all your unsaved changes will be lost. You can try and refresh this page to resolve the problem.", "wordpress-seo" ) }</p>
			<div className="yst-flex yst-gap-2">
				<Button onClick={ handleRefreshClick }>{ __( "Refresh this page", "wordpress-seo" ) }</Button>
			</div>
		</div>
	);
};

RouteErrorFallback.propTypes = {
	error: PropTypes.object.isRequired,
};

export default RouteErrorFallback;
