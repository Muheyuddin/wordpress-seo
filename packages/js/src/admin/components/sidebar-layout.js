import PropTypes from "prop-types";

/**
 * @param {JSX.Element} children The main content.
 * @param {JSX.Element} sidebar The sidebar content.
 * @returns {JSX.Element} The layout.
 */
const SidebarLayout = ( { children, sidebar } ) => (
	<div className="yst-p-4 min-[783px]:yst-p-8 yst-flex yst-gap-4">
		<aside className="yst-sidebar yst-sidebar-nav yst-shrink-0 yst-hidden min-[783px]:yst-block yst-pb-6 yst-bottom-0 yst-w-56">
			{ sidebar }
		</aside>
		<div className="yst-flex yst-grow yst-flex-wrap">
			<div className="yst-grow yst-space-y-6 yst-mb-8 xl:yst-mb-0">
				<main className="yst-rounded-lg yst-bg-white yst-shadow">
					{ children }
				</main>
			</div>
		</div>
	</div>
);

SidebarLayout.propTypes = {
	children: PropTypes.node.isRequired,
	sidebar: PropTypes.node.isRequired,
};

export default SidebarLayout;
