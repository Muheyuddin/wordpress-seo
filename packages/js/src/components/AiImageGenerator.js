import SlotWithDefault from "./slots/SlotWithDefault";

/**
 * The AiImageGenerator component.
 *
 * @returns {wp.Element} The AiImageGenerator component.
 */
export const AiImageGenerator = () => {
	return (
		<SlotWithDefault  name="AiImageGenerator">
			<p> Premium needed</p>
		</SlotWithDefault>
	);
};
