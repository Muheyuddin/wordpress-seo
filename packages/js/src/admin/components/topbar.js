import { __ } from "@wordpress/i18n";
import { TopbarNavigation, useSvgAria } from "@yoast/ui-library";
import classNames from "classnames";
import { Link, useLocation } from "react-router-dom";
import { YoastLogo } from "../../settings/components";
import { useSelectAdmin } from "../hooks";

/**
 * @returns {JSX.Element} The topbar.
 */
const Topbar = () => {
	const isPremium = useSelectAdmin( "selectFromShared", [], "isPremium", false );
	const links = useSelectAdmin( "selectLinksFromRoutes" );
	const svgAriaProps = useSvgAria();
	const { pathname } = useLocation();

	return (
		<TopbarNavigation activePath={ pathname }>
			<TopbarNavigation.Container>
				<TopbarNavigation.Topbar>
					<TopbarNavigation.Start>
						<TopbarNavigation.MobileMenuButton
							openButtonScreenReaderText={ __( "Open top bar navigation menu", "wordpress-seo" ) }
							closeButtonScreenReaderText={ __( "Close top bar navigation menu", "wordpress-seo" ) }
						/>
					</TopbarNavigation.Start>

					<TopbarNavigation.Center>
						<TopbarNavigation.LogoContainer>
							<Link
								id="link-yoast-logo-topbar"
								to="/"
								className={ classNames(
									"yst-block yst-py-2.5 yst-border-b-2 yst-border-transparent focus:yst-ring-primary-500",
									pathname === "/" && "yst-border-primary-500"
								) }
								aria-label={ isPremium ? "Yoast SEO Premium" : "Yoast SEO" }
							>
								<YoastLogo className="yst-w-40" { ...svgAriaProps } />
							</Link>
						</TopbarNavigation.LogoContainer>
						<TopbarNavigation.LinksContainer>
							{ links.map( ( { children, ...props }, index ) => (
								<TopbarNavigation.Link
									key={ `link-${ index }` }
									as={ Link }
									hrefProp="to"
									{ ...props }
								>
									{ children }
								</TopbarNavigation.Link>
							) ) }
						</TopbarNavigation.LinksContainer>
					</TopbarNavigation.Center>
				</TopbarNavigation.Topbar>

				<TopbarNavigation.MobileContainer>
					{ links.map( ( { children, ...props }, index ) => (
						<TopbarNavigation.MobileLink
							key={ `link-mobile-${ index }` }
							as={ Link }
							hrefProp="to"
							{ ...props }
						>
							{ children }
						</TopbarNavigation.MobileLink>
					) ) }
				</TopbarNavigation.MobileContainer>
			</TopbarNavigation.Container>
		</TopbarNavigation>
	);
};

export default Topbar;
