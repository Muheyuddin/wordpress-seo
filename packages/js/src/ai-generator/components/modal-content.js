import { useDispatch, useSelect } from "@wordpress/data";
import { useMemo } from "@wordpress/element";
import { AiGenerateTitlesAndDescriptionsUpsell } from "../../shared-admin/components";

const STORE = "yoast-seo/editor";

/**
 * @returns {JSX.Element} The element.
 */
export const ModalContent = () => {
	const learnMoreLink = useSelect( select => select( STORE ).selectLink( "https://www.yoa.st/ai-generator-learn-more" ), [] );
	const upsellLink = useSelect( select => select( STORE ).selectLink( "https://yoa.st/ai-generator-upsell" ), [] );

	const imageLink = useSelect( select => select( STORE ).selectImageLink( "ai-generator-preview.png" ), [] );
	const thumbnail = useMemo( () => ( {
		src: imageLink,
		width: "432",
		height: "244",
	} ), [ imageLink ] );

	const value = useSelect( select => select( STORE ).selectWistiaEmbedPermissionValue(), [] );
	const status = useSelect( select => select( STORE ).selectWistiaEmbedPermissionStatus(), [] );
	const { setWistiaEmbedPermission: set } = useDispatch( STORE );
	const wistiaEmbedPermission = useMemo( () => ( { value, status, set } ), [ value, status, set ] );

	return (
		<AiGenerateTitlesAndDescriptionsUpsell
			learnMoreLink={ learnMoreLink }
			upsellLink={ upsellLink }
			thumbnail={ thumbnail }
			wistiaEmbedPermission={ wistiaEmbedPermission }
		/>
	);
};
