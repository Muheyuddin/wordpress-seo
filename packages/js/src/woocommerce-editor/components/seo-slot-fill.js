import { createSlotFill } from "@wordpress/components";
import sortComponentsByRenderPriority from "../../helpers/sortComponentsByRenderPriority";

const SLOT_FILL_NAME = "yoast-seo/woocommerce-editor/seo";

const { Slot, Fill } = createSlotFill( SLOT_FILL_NAME );

/**
 * @returns {JSX.Element} The element.
 */
export const SeoSlot = () => (
	<Slot>
		{ ( fills ) => sortComponentsByRenderPriority( fills ) }
	</Slot>
);

export const SeoFill = Fill;
