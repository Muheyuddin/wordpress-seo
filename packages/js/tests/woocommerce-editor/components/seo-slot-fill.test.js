import { Fill, SlotFillProvider } from "@wordpress/components";
import SidebarItem from "../../../src/components/SidebarItem";
import { SeoFill, SeoSlot } from "../../../src/woocommerce-editor/components/seo-slot-fill";
import { render, screen } from "../../test-utils";

describe( "SeoSlotFill", () => {
	it( "fills the slot", () => {
		render( <SlotFillProvider>
			<SeoSlot />
			<SeoFill>Foo</SeoFill>
		</SlotFillProvider> );

		expect( screen.getByText( "Foo" ) ).toBeInTheDocument();
	} );

	it( "fills the slot when using the name", () => {
		render( <SlotFillProvider>
			<SeoSlot />
			<Fill name="yoast-seo/woocommerce-editor/seo">Foo</Fill>
		</SlotFillProvider> );

		expect( screen.getByText( "Foo" ) ).toBeInTheDocument();
	} );

	it( "sorts the fills by renderPriority", () => {
		render( <SlotFillProvider>
			<div data-testid={ 1 }>
				<SeoSlot />
				<SeoFill>
					<SidebarItem renderPriority={ 200 }>200</SidebarItem>
					<SidebarItem renderPriority={ 10 }>10</SidebarItem>
				</SeoFill>
			</div>
		</SlotFillProvider> );

		expect( screen.getByText( "200" ) ).toBeInTheDocument();
		expect( screen.getByText( "10" ) ).toBeInTheDocument();
		expect( screen.getByTestId( 1 ) ).toMatchSnapshot();
	} );
} );
