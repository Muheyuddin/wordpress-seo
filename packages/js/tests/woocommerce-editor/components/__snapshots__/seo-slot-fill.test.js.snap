// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeoSlotFill sorts the fills by renderPriority 1`] = `
<div
  data-testid="1"
>
  <div>
    10
  </div>
  <div>
    200
  </div>
</div>
`;
